etcdserver/v3_server.go

```diff
@@ -551,6 +507,28 @@ func (s *EtcdServer) authInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error
 	return authInfo, nil
 }
 
+func (s *EtcdServer) doSerialize(ctx context.Context, chk func(*auth.AuthInfo) error, get func()) error {
+	var ai *auth.AuthInfo
+	for {
+		authInfo, err := s.authInfoFromCtx(ctx)
+		if err != nil {
+			return err
+		}
+		if ai = authInfo; ai == nil {
+			ai = &auth.AuthInfo{}
+		}
+		if err = chk(ai); err != nil {
+			return err
+		}
+		get()
```

Comment:
this line of code is not easy to understand. probably we need add some comments to the doSerialize func at least.